WcaOnRails/app/controllers/accounts/registrations_controller.rb

```diff
@@ -1,7 +1,20 @@
 # frozen_string_literal: true
 
 class Accounts::RegistrationsController < Devise::RegistrationsController
+  prepend_before_action :check_captcha, only: [:create]
   protected def after_update_path_for(resource)
     edit_user_registration_path
   end
+
+  private
+
+  def check_captcha
+    return if verify_recaptcha
+    self.resource = resource_class.new sign_up_params
+    set_minimum_password_length
```

Comment:
We need to validate the resource (user) in order to show the validation messages ([reference](https://github.com/heartcombo/devise/wiki/How-To:-Use-Recaptcha-with-Devise#deviseregistrationscontroller)).

```suggestion
    self.resource = resource_class.new sign_up_params
    resource.validate
    set_minimum_password_length
```

Reply:
Ok, gotcha!